chia/consensus/blockchain.py

```diff
@@ -420,13 +431,41 @@ async def _reconsider_peak(
 
                     # Set additions first, then removals in order to handle ephemeral coin state
                     # Add in height order is also required
+                    coin_subscriptions, ph_subscriptions = subscriptions
                     record: Optional[CoinRecord]
                     for record in added_rec:
                         assert record
                         lastest_coin_state[record.name] = record
+                        if record.name in coin_subscriptions:
+                            subscribed_peers = coin_subscriptions[record.name]
+                            for peer in subscribed_peers:
+                                if peer not in changes_for_peer:
+                                    changes_for_peer[peer] = set()
+                                changes_for_peer[peer].add(record.coin_state)
```

Comment:
I think it would make sense to factor out the subscription management to a separate function. It would give it a name, and deduplicate some of this code, and make the main purpose of the logic here less diluted

Reply:
Gotcha